lib/spreedly/environment.rb

```diff
@@ -285,7 +285,7 @@ def add_extra_options_for_basic_ops(doc, options)
       add_shipping_address_override(doc, options)
       add_to_doc(doc, options, :order_id, :description, :ip, :email, :merchant_name_descriptor,
                                :merchant_location_descriptor, :redirect_url, :callback_url,
-                               :continue_caching)
+                               :continue_caching, :attempt_3dsecure, :browser_info)
```

Comment:
Should `:required_action`, `:challenge_form`, `:challenge_url`, and `:device_fingerprint_form` be add here as well?

Reply:
If I'm understanding the code paths right, I don't think so. :required_action, :challenge_form, :challenge_url, and :device_fingerprint_form are all readonly attributes.

Reply:
Perfect 👍